vscode-lean4/src/infoview.ts

```diff
@@ -225,26 +219,26 @@ export class InfoProvider implements Disposable {
                 this.webviewPanel = undefined;
             });
             this.webviewPanel = webviewPanel;
-            webviewPanel.webview.html = this.initialHtml();
+            webviewPanel.webview.html = this.initialHtml()
 
-            // this.client.initializeResult can be null if the server failed to start.
-            if (this.client.initializeResult) {
-                await webviewPanel.api.initialize(this.client.initializeResult, this.getLocation(editor))
-            }
+            await this.initWebview(editor)
         }
+    }
+
+    /** Start executing the infoview UI scripts and send over editor information
+     * such as the cursor position. Restart and reinit the UI if the infoview
+     * is already running. Do nothing if there is no webview panel. */
+    private async initWebview(editor: TextEditor) {
+        if (!this.webviewPanel) return
 
+        // this.client.initializeResult can be null if the server failed to start.
+        // in this case the infoview displays an error message.
+        await this.webviewPanel.api.initialize(this.client.initializeResult, this.getLocation(editor))
         // The infoview listens for server notifications such as diagnostics passively,
         // so when it is first started we must re-send those as if the server did.
-        await this.sendPosition();
```

Comment:
I am not sure why we were re-sending this on revealing the infoview given that even when hidden the infoview should still receive position updates etc, but marking this in case I am missing something.